### ISSUE TITLE
trial.suggest fails after n_startup_trials when using TPESampler and tensorflow model

### ISSUE THREAD

**Issue by satulovsky** [optuna/optuna#3344]
### Expected behavior

First of all, thank you for this awesome library!

I am building a class to hyper-tune a machine learning model:

```python
class Hypertune:

    def __init__(self, objective_func: Callable, n_trials: int = 50):
        self.trials_ = None
        self.best_parameters_ = None
        self.best_value_ = None
        self.n_trials = n_trials
        self.study = optuna.create_study(direction='minimize', sampler=TPESampler(n_startup_trials=29))
        self.objective = objective_func

```

Initially, everything goes as expected when I run:

```python
self.study.optimize(self.objective, n_trials=self.n_trials)
```

However I get a TypeError as soon as the number of trials goes beyond `n_startup_trials` from the TPEsampler (29 in this case, I also tried other values)

I have used the same code successfully in another machine learning project (I was using older version of python and of the libraries reported here, though)

### Environment

- Optuna version: 3.0.0a2
- Python version: 3.10.2
- OS: Linux-5.11.0-1028-aws-x86_64-with-glibc2.31
- (Optional) Other libraries and their versions:
numpy 1.22.2
pandas 1.4.1 
tensorflow 2.8.0
keras 2.8.0

### Error messages, stack traces, or logs

```shell
Trial 29 failed because of the following error: TypeError("'NoneType' object is not callable")
Traceback (most recent call last):
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/numpy/core/getlimits.py", line 459, in __new__
    dtype = numeric.dtype(dtype)
TypeError: 'NoneType' object is not callable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/study/_optimize.py", line 213, in _run_trial
    value_or_values = func(trial)
  File "/home/ubuntu/Pall_temp/src/hypertune/bug_report.py", line 49, in objective
    params = set_parameters(trial)
  File "/home/ubuntu/Pall_temp/src/hypertune/bug_report.py", line 65, in set_parameters
    params['models']['dense']['layer1'] = trial.suggest_int('layer1', 10, 500)
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/trial/_trial.py", line 452, in suggest_int
    return int(self._suggest(name, distribution))
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/trial/_trial.py", line 719, in _suggest
    param_value = self.study.sampler.sample_independent(
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/samplers/_tpe/sampler.py", line 442, in sample_independent
    samples_below = mpe_below.sample(self._rng, self._n_ei_candidates)
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/samplers/_tpe/parzen_estimator.py", line 99, in sample
    active = rng.choice(len(self._weights), size, p=self._weights)
  File "mtrand.pyx", line 920, in numpy.random.mtrand.RandomState.choice
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/numpy/core/getlimits.py", line 462, in __new__
    dtype = numeric.dtype(type(dtype))
TypeError: 'NoneType' object is not callable
Traceback (most recent call last):
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/numpy/core/getlimits.py", line 459, in __new__
    dtype = numeric.dtype(dtype)
TypeError: 'NoneType' object is not callable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/study/_optimize.py", line 66, in _optimize
    _optimize_sequential(
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/study/_optimize.py", line 163, in _optimize_sequential
    trial = _run_trial(study, func, catch)
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/study/_optimize.py", line 264, in _run_trial
    raise func_err
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/study/_optimize.py", line 213, in _run_trial
    value_or_values = func(trial)
  File "/home/ubuntu/Pall_temp/src/hypertune/bug_report.py", line 49, in objective
    params = set_parameters(trial)
  File "/home/ubuntu/Pall_temp/src/hypertune/bug_report.py", line 65, in set_parameters
    params['models']['dense']['layer1'] = trial.suggest_int('layer1', 10, 500)
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/trial/_trial.py", line 452, in suggest_int
    return int(self._suggest(name, distribution))
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/trial/_trial.py", line 719, in _suggest
    param_value = self.study.sampler.sample_independent(
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/samplers/_tpe/sampler.py", line 442, in sample_independent
    samples_below = mpe_below.sample(self._rng, self._n_ei_candidates)
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/optuna/samplers/_tpe/parzen_estimator.py", line 99, in sample
    active = rng.choice(len(self._weights), size, p=self._weights)
  File "mtrand.pyx", line 920, in numpy.random.mtrand.RandomState.choice
  File "/home/ubuntu/.pyenv/versions/3.10.2/lib/python3.10/site-packages/numpy/core/getlimits.py", line 462, in __new__
    dtype = numeric.dtype(type(dtype))
TypeError: 'NoneType' object is not callable

Process finished with exit code 1
```

### Steps to reproduce

(I am omitting the code for CustomDataGen, Features and Models as they are proprietary)

---

```python
from multiprocessing import cpu_count
from typing import Callable
import yaml

import optuna
from addict import Dict
from optuna.samplers import TPESampler

from src.data.data_generator import CustomDataGen
from src.data.features import Features
from src.models.models import Models


class Hypertune:

    def __init__(self, objective_func: Callable, n_trials: int = 50):
        self.trials_ = None
        self.best_parameters_ = None
        self.best_value_ = None
        self.n_trials = n_trials
        self.study = optuna.create_study(direction='minimize', sampler=TPESampler(n_startup_trials=29))
        self.objective = objective_func

    def optimize_(self):
        self.study.optimize(self.objective, n_trials=self.n_trials)
        self.best_parameters_ = self.study.best_params
        self.best_value_ = self.study.best_value
        self.trials_ = self.study.trials

    @property
    def best_params(self):
        if not self.best_parameters_:
            self.optimize_()
        return self.best_parameters_

    @property
    def best_value(self):
        if not self.best_value_:
            self.optimize_()
        return self.best_value_


if __name__ == '__main__':

    def objective(trial: optuna.Trial) -> float:
        params = set_parameters(trial)
        model = Models(params).dense()
        train_gen = CustomDataGen(Features().data,
                                  batch_size=params.train.train_batch_size,
                                  ar_order=params.data_prep.ar_order)
        history = model.fit(train_gen,
                            epochs=params.train.epochs, verbose=params.train.verbose,
                            workers=cpu_count() - 4, use_multiprocessing=True)
        return history.history['mae'][-1]


    def set_parameters(trial: optuna.Trial) -> Dict:
        """ Overrides parameters in optuna.yml file with parameters suggested by the trial """
        params_relative_path = '../models/optuna.yml'
        with open(params_relative_path, 'r') as f:
            params = yaml.safe_load(f)
            params['models']['dense']['layer1'] = trial.suggest_int('layer1', 10, 500)
            params['models']['dense']['dropout1'] = trial.suggest_uniform('dropout1', 0, 1)
            params['models']['dense']['layer2'] = trial.suggest_int('layer2', 10, 500)
            params['models']['dense']['dropout2'] = trial.suggest_uniform('dropout2', 0, 1)
            params['models']['dense']['layer3'] = trial.suggest_int('layer3', 10, 500)
            params['models']['dense']['dropout3'] = trial.suggest_uniform('dropout3', 0, 1)
        return Dict(params)

    hypertune = Hypertune(objective, n_trials=200)
    best_value = hypertune.best_value
    best_params = hypertune.best_params

    print(f'Best value: {best_value}')
    print(f'Best parameters: {best_params}')
```

### Additional context (optional)

Using the libraries described here I can run the minimalistic example from the Optuna tutorial, the TypeError happens when I am fitting the tensorflow model at each objective evaluation.

```python
import optuna
from optuna.samplers import TPESampler


class Objective(object):
    def __init__(self, min_x, max_x):
        # Hold this implementation specific arguments as the fields of the class.
        self.min_x = min_x
        self.max_x = max_x

    def __call__(self, trial):
        # Calculate an objective value by using the extra arguments.
        x = trial.suggest_loguniform("x", self.min_x, self.max_x)
        return (x - 2) ** 2

# Execute an optimization by using an `Objective` instance.
study = optuna.create_study(sampler=TPESampler(n_startup_trials=29))
study.optimize(Objective(0.001, 0.1), n_trials=100)
```

**Comment by nzw0301** [comment]
Hi, could you share minimal reproducible codes with us since `CustomDataGen`, `Features`, and `Models` are not defined?

**Comment by satulovsky** [comment]
I will try to generate something this week!

**Comment by nzw0301** [comment]
Awesome! Thank you!

**Comment by himkt** [comment]
@satulovsky Does this also happen with Python 3.9? I just found https://youtrack.jetbrains.com/issue/PY-52137 which might be related.

**Comment by nzw0301** [comment]
@himkt  Thank you for sharing it!

I should have mentioned that Optuna does not support Python 3.10 yet; Optuna with python 3.8 pr 3.9 might be reliable.

**Comment by satulovsky** [comment]
@himkt Thank you so much!!  You nailed it, that was it ...

After building a reproducible example that showed the error in debug mode, I tried it without the debugger and it ran fine. Then I ran it in python 3.9, as you suggested, and it ran fine even in debugging mode.

@nzw0301 Thank you for the suggestion, I will have to downgrade to python 3.9 since I am addicted to using the debugger. No pattern matching for me yet ... :-(
